### PR TITLE
test/e2e: fix for stdin error on cleanup

### LINKED ISSUE #31884: [k8s.io] Kubectl client [k8s.io] Guestbook application should create and stop a working application [Conformance] {Kubernetes e2e suite}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/31878/kubernetes-pull-build-test-e2e-gke/10546/

Failed: [k8s.io] Kubectl client [k8s.io] Guestbook application should create and stop a working application [Conformance] {Kubernetes e2e suite}

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:267
Expected error:
    <exec.CodeExitError>: {
        Err: {
            s: "error running &{/workspace/kubernetes/platforms/linux/amd64/kubectl [kubectl --server=https://130.211.127.70 --kubeconfig=/workspace/.kube/config create -f - --namespace=e2e-tests-kubectl-gfgao] []  0xc820558c00  Error from server: an error on the server (\"Internal Server Error: \\\"/apis/batch/v1\\\"\") has prevented the request from succeeding\n [] <nil> 0xc820559560 exit status 1 <nil> true [0xc8200a6bf8 0xc8200a6c20 0xc8200a6c30] [0xc8200a6bf8 0xc8200a6c20 0xc8200a6c30] [0xc8200a6c00 0xc8200a6c18 0xc8200a6c28] [0xadd7c0 0xadd920 0xadd920] 0xc820a201e0}:\nCommand stdout:\n\nstderr:\nError from server: an error on the server (\"Internal Server Error: \\\"/apis/batch/v1\\\"\") has prevented the request from succeeding\n\nerror:\nexit status 1\n",
        },
        Code: 1,
    }
    error running &{/workspace/kubernetes/platforms/linux/amd64/kubectl [kubectl --server=https://130.211.127.70 --kubeconfig=/workspace/.kube/config create -f - --namespace=e2e-tests-kubectl-gfgao] []  0xc820558c00  Error from server: an error on the server ("Internal Server Error: \"/apis/batch/v1\"") has prevented the request from succeeding
     [] <nil> 0xc820559560 exit status 1 <nil> true [0xc8200a6bf8 0xc8200a6c20 0xc8200a6c30] [0xc8200a6bf8 0xc8200a6c20 0xc8200a6c30] [0xc8200a6c00 0xc8200a6c18 0xc8200a6c28] [0xadd7c0 0xadd920 0xadd920] 0xc820a201e0}:
    Command stdout:
    
    stderr:
    Error from server: an error on the server ("Internal Server Error: \"/apis/batch/v1\"") has prevented the request from succeeding
    
    error:
    exit status 1
    
not to have occurred
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2121
```

Happened on a presubmit run in #31878.

Previous issues for this test: #26175 #26846 #27334 #28293 #29149

### PR DESCRIPTION
**Pull request by jessfraz**
**What this PR does / why we need it**: helps with flake in #31884 

**Which issue this PR fixes** _(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR gets merged)_: fixes #31884 

**Special notes for your reviewer**:

**Release note**:

``` release-note
NONE
```

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32310)

### CODE REVIEW

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -160,6 +160,23 @@ func runKubectlRetryOrDie(args ...string) string {
 	return output
 }
 
+func runKubectlInputRetryOrDie(data string, args ...string) string {
+	var err error
+	var output string
+	for i := 0; i < 3; i++ {
+		output, err = framework.NewKubectlCommand(args...).WithStdinData(data).Exec()
+		if err == nil || !strings.Contains(err.Error(), registry.OptimisticLockErrorMsg) {
```
Comment (janetkuo) [inline test/e2e/kubectl.go]: `errors.IsConflict(err)`

**Comment by janetkuo** [comment]
So this is retrying on conflict when `kubectl delete`? Which failure in #31884 is caused by this?

**Comment by jfrazelle** [comment]
It was an attempt to fix 
```
error when stopping "STDIN": an error on the server ("Internal Server Error: \"/apis/extensions/v1beta1/namespaces/e2e-tests-kubectl-p56l7/deployments/redis-master\"") has prevented the request from succeeding (get deployments.extensions redis-master)
```

**Comment by janetkuo** [comment]
I don't think it's caused by conflict err since deployment reaper retries on conflict https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/stop.go#L471-L489

The error message isn't helpful. We'll need api server log to debug more...

**Comment by jfrazelle** [comment]
Oh so I wasn't really concerned with the error I was more trying to just
have it try again, but ya the error is curious

**Comment by jessfraz** [comment]
closing, I don't think this is the correct fix

**Pull request closed without merging**